### PR TITLE
[DO NOT MERGE]Training DLC for Pytorch 1.10

### PR DESCRIPTION
**Pull request by waytrue17**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron"/"graviton">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arjkesh** [comment]
@waytrue17 - the EC2 NCCL tests seem to be failing on the gpu training image.

There are also some pip check and torch vision failures in the sanity tests - these seem to be related to the fact that we don't have a TV version that supports PT1.10 yet

File: pytorch/training/docker/1.10/py3/Dockerfile.diy.cpu [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ && rm -rf openmpi-$OPEN_MPI_VERSION
+
+# The ENV variables declared below are changed in the previous section
+# Grouping these ENV variables in the first section causes
+# ompi_info to fail. This is only observed in CPU containers
+ENV PATH="$PATH:/home/.openmpi/bin"
+ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/home/.openmpi/lib/"
+RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value
+
+# Copy workaround script for incorrect hostname
```
Comment (arjkesh) [inline pytorch/training/docker/1.10/py3/Dockerfile.diy.cpu:72]: this looks specific to SM images - we can prob move this along with the custom entrypoint to the SM image as a result

File: pytorch/training/docker/1.10/py3/cu113/Dockerfile.diy.gpu [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    pandas \
+    h5py \
+    requests \
+    libgcc \
+ && conda clean -ya
+
+WORKDIR /opt/pytorch
+
+# Copy workaround script for incorrect hostname
+COPY changehostname.c /
```
Comment (arjkesh) [inline pytorch/training/docker/1.10/py3/cu113/Dockerfile.diy.gpu:169]: this looks specific to SM images - we can prob move this along with the custom entrypoint to the SM image as a result

File: test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py [hunk 3]
```diff
@@ -84,6 +87,9 @@ def test_pytorch_train_dgl_gpu(pytorch_training, ec2_connection, gpu_only, py3_o
     execute_ec2_training_test(ec2_connection, pytorch_training, PT_DGL_CMD)
 
 
+# TODO: unskip this test when DGL supports PT 1.10. If we need to release without DGL, make sure this skip
+# TODO: is conditional for PT1.10, so that DGL tests still run on prior images.
+@pytest.mark.skip(reason="Skip DGL tests for PT1.10, since there is no compatible version at the moment")
```
Comment (arjkesh) [inline test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py:92]: unskip or conditionally skip before merge

File: test/dlc_tests/ecs/pytorch/training/test_ecs_pytorch_training.py [hunk 4]
```diff
@@ -105,6 +105,9 @@ def test_ecs_pytorch_s3_plugin_training_gpu(gpu_only, ecs_container_instance, py
                                          num_gpus=num_gpus)
 
 
+# TODO: unskip this test when DGL supports PT 1.10. If we need to release without DGL, make sure this skip
+# TODO: is conditional for PT1.10, so that DGL tests still run on prior images.
+@pytest.mark.skip(reason="Skip DGL tests for PT1.10, since there is no compatible version at the moment")
```
Comment (arjkesh) [inline test/dlc_tests/ecs/pytorch/training/test_ecs_pytorch_training.py:110]: unskip or conditionally skip before merge

File: test/dlc_tests/ecs/pytorch/training/test_ecs_pytorch_training.py [hunk 5]
```diff
@@ -128,6 +131,9 @@ def test_ecs_pytorch_training_dgl_cpu(cpu_only, py3_only, ecs_container_instance
     ecs_utils.ecs_training_test_executor(ecs_cluster_name, cluster_arn, training_cmd, pytorch_training, instance_id)
 
 
+# TODO: unskip this test when DGL supports PT 1.10. If we need to release without DGL, make sure this skip
+# TODO: is conditional for PT1.10, so that DGL tests still run on prior images.
+@pytest.mark.skip(reason="Skip DGL tests for PT1.10, since there is no compatible version at the moment")
```
Comment (arjkesh) [inline test/dlc_tests/ecs/pytorch/training/test_ecs_pytorch_training.py:136]: unskip or conditionally skip before merge

File: test/dlc_tests/eks/pytorch/training/test_eks_pytorch_training.py [hunk 6]
```diff
@@ -156,6 +156,9 @@ def test_eks_pt_s3_plugin_single_node_training(pytorch_training, pt17_and_above_
         run("kubectl delete pods {}".format(pod_name))
 
 
+# TODO: unskip this test when DGL supports PT 1.10. If we need to release without DGL, make sure this skip
+# TODO: is conditional for PT1.10, so that DGL tests still run on prior images.
+@pytest.mark.skip(reason="Skip DGL tests for PT1.10, since there is no compatible version at the moment")
```
Comment (arjkesh) [inline test/dlc_tests/eks/pytorch/training/test_eks_pytorch_training.py:161]: unskip or conditionally skip before merge

**Review by arjkesh — CHANGES_REQUESTED** [review]

**Comment by Kyeongpil** [comment]
Is there any progress? I want to know when we can use the Pytorch 1.10 container on Sagemaker.

**Comment by waytrue17** [comment]
Close this PR as it was addressed in other PRs

**Comment by waytrue17** [comment]
> Hi, @waytrue17. Is there any progress? I want to know when we can use the Pytorch 1.10 container on Sagemaker.

Hi Kyeongpil, the PT 1.10 DLC effort has been done in other PRs #1459 #1482. I think it will come soon as our team is actively pushing the release

**Pull request closed without merging**